pkg/controller/servicemesh/controlplane/hooks.go

```diff
@@ -68,134 +53,9 @@ func (r *ControlPlaneReconciler) processNewObject(object *unstructured.Unstructu
 }
 
 func (r *ControlPlaneReconciler) processDeletedObject(object *unstructured.Unstructured) error {
-	switch object.GetKind() {
-	case "Route":
-		if object.GetName() == "kiali" {
-			return r.processDeletedKialiRoute(object)
-		}
-	}
 	return nil
 }
 
-var (
-	grafanaRegexp = regexp.MustCompile("(grafana:\\s*url:).*?\n")
-	jaegerRegexp  = regexp.MustCompile("(jaeger:\\s*url:).*?\n")
-)
-
-func (r *ControlPlaneReconciler) patchKialiConfig(object *unstructured.Unstructured) error {
```

Comment:
Don't we still need to do this, so Kiali has the correct route names?

Reply:
Kiali now has auto-detection via routes built into the server. These settings in the Kiali CR are still there for people that have custom URLs that cannot be autodetected... but this auto-detection by looking at the routes is now down inside the Kiali Server so we don't need to do it here. We just leave the settings unset or empty and that tells Kiali to auto-detect. See:

https://github.com/kiali/kiali/pull/1022

Reply:
does it look for a route named `jaeger` or `tracing`, or does it just use the internal services?

Reply:
By default (if you don't tell it what service it is), it will attempt to look for either jaeger-query or tracing.. see:

https://github.com/kiali/kiali/blob/master/status/discover.go#L18

Here is where it does the lookup:

https://github.com/kiali/kiali/blob/master/status/discover.go#L98-L110

Reply:
So that means it will default to tracing.  Currently, that route is in flux, and jaeger exposes the jaeger-query service through a route named `jaeger`.

Reply:
We can tell @aljesusg if there is a problem we find in the kiali auto-detection. Or if we need to change that algorithm to look for different names.

Reply:
From Gary Brown of the Jaeger team:

> the 'tracing' service is intended as a tracing
> provider independent UI endpoint, so users did not have to be aware of the
> implementation being used. As service mesh will only support Jaeger, I
> think the 'tracing' service is not relevant.

Then perhaps we should remove or change the order here (you see here the tracing route is looked at first when Kiali is auto-detecting the jaeger endpoint).

https://github.com/kiali/kiali/blob/master/status/discover.go#L18

If tracing is not relevant when a mesh only supports Jaeger, then we should be looking for the jaeger route first, and only then look for tracing if we don't find it.

@aljesusg

Reply:
Hi, Kiali looks in a list of services that we set in the code if the user not set the tracing service in the configmap.

```go
var tracingLookupRoutes = [...]string{"tracing", "jaeger-query"}
```

If there is a service in the Istio namespace installation with this name kiali will assume it as the tracing service, after that kiali looks for the route if the environment is openshift.

So we can change the order.

Reply:
I think we should keep this code, but adapt it to modify the kiali CR.  This is where you could disable missing dependencies (e.g. jaeger, grafana).

Reply:
we can't disable missing grafana today - kiali needs this: see https://issues.jboss.org/browse/KIALI-3066

Reply:
What I am going to do is - I will try to put this code back - but obviously change it so it modifies the Kiali CR (this code here is modifying the configmap which we should not be doing anymore since the kiali operator will deal with configmap updates).

I will have this code set the "external_services.tracing.enabled" flag to false if Jaeger is not available. That is a valid setting.

I will have this code set the "external_services.grafana.enabled" flag to false if Grafana is not available. This is not used today - its just a flag that will be ignored, but once https://issues.jboss.org/browse/KIALI-3066 is implemented, it will take effect. Kiali won't burp if we give it an unknown setting, it will just blindly ignore it.

Reply:
Sounds good.  I think it's probably safe to assume that if the routes are missing, the functionality is missing too.